lib/tilex/posts.ex

```diff
@@ -35,15 +35,14 @@ defmodule Tilex.Posts do
 
   def random_post_by_channel(channel_name) do
     channel = Repo.get_by!(Channel, name: channel_name)
-
+    channel_id = channel.id
     posts_count = 1
 
     query =
       from(
         post in Post,
         order_by: fragment("random()"),
-        join: channel in assoc(post, :channel),
-        where: channel.name == ^channel_name,
+        where: post.channel_id == ^channel_id,
```

Comment:
Given it's `limit: 1`, would it be better to leave the join, remove the separate Channel DB `get_by!/2` call and just return the single posts channel? Then you'd only have a single DB call. ie something like::

```elixir
 def random_post_by_channel(channel_name) do
    posts_count = 1

    query =
      from(
        post in Post,
        order_by: fragment("random()"),
        join: channel in assoc(post, :channel),
        where: channel.name == ^channel_name,
        limit: ^posts_count,
        preload: [:channel, :developer]
      )
    post = Repo.one(query)

    {[post], posts_count, post.channel}
  end
```

Reply:
that's even better, thanks @tmock12 !